lib/ui/painting/canvas.cc

```diff
@@ -441,8 +441,16 @@ void Canvas::drawImage(const CanvasImage* image,
     builder()->drawImage(image->image(), SkPoint::Make(x, y), sampling,
                          with_attributes);
   } else if (canvas_) {
+    auto dl_image = image->image();
+    if (!dl_image) {
+      return;
+    }
+    auto sk_image = dl_image->skia_image();
+    if (!sk_image) {
+      return;
+    }
```

Comment:
looks like food for a helper method?

Also, all Skia drawImageish methods do a RETURN_ON_NULL(image) at the top which might help simplify these sequences...?